tests/requests_tests/tests.py

```diff
@@ -737,6 +737,30 @@ def test_POST_multipart_with_file(self):
         self.assertEqual(len(request.FILES), 1)
         self.assertIsInstance((request.FILES["File"]), InMemoryUploadedFile)
 
+    def test_POST_multipart_with_empty_filename(self):
+        payload = FakePayload(
+            "\r\n".join(
+                [
+                    f"--{BOUNDARY}",
+                    'Content-Disposition: form-data; name="File"; filename=""',
+                    "Content-Type: application/octet-stream",
+                    "",
+                    "DATA",
+                    f"--{BOUNDARY}--",
+                ]
+            )
+        )
+        request = WSGIRequest(
+            {
+                "REQUEST_METHOD": "POST",
+                "CONTENT_TYPE": MULTIPART_CONTENT,
+                "CONTENT_LENGTH": len(payload),
+                "wsgi.input": payload,
+            }
+        )
+        self.assertEqual(len(request.FILES), 1)
+        self.assertIsInstance((request.FILES["File"]), InMemoryUploadedFile)
+
```

Comment:
thought: I'd add a case what happens if hit the jackpot and the random filename already exists 🤔